consensus/polybft/fsm.go

```diff
@@ -517,13 +514,11 @@ func (f *fsm) verifyValidatorsUptimeTx(transactions []*types.Transaction) error
 				createdUptimeTx.Hash,
 			)
 		}
-	} else {
-		if blockUptimeTx != nil && blockUptimeTx.Hash == createdUptimeTx.Hash {
-			return errors.New("didn't expect uptime transaction in the middle of an epoch")
-		}
+
+		return nil
 	}
 
-	return nil
+	return errors.New("didn't expect uptime transaction in a non ending epoch block")
```

Comment:
This could go to `var` at the beginning where we do errors declaration

Reply:
Fix: https://github.com/0xPolygon/polygon-edge/pull/1186/commits/3c3920420847ec1cb1b83e8ff589b424dee653d8